### PR TITLE
Create a common controller config

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

**Comment by kargakis** [comment]
@spxtr @cjwagner I would like to gather thoughts on whether this is a viable approach for extending prow config. Also, we may want to do something similar for external plugins (for example I want to pass a whitelist of base and target branches in the cherrypicker: https://github.com/kubernetes/test-infra/pull/5010)

File: prow/config/config.go [hunk 1]
```diff
@@ -270,21 +246,6 @@ func parseConfig(c *Config) error {
 		return fmt.Errorf("plank has invalid max_concurrency (%d), it needs to be a non-negative number", c.Plank.MaxConcurrency)
 	}
```
Comment (cjwagner) [inline prow/config/config.go]: The validation for the plank config looks identical to that of the jenkins-operator.
Can you move this to a `ValidateController` function that can be used on `c.Plank` here and also called from `jenkins/config.go`?

**Comment by kargakis** [comment]
I am creating just the common struct and validation helper for now, ptal

**Pull request merged**